File: src/sdk/python/ainari_sdk/ainari_sdk/quota.py
# Copyright 2022-2026 Tobias Anker <tobias.anker@kitsunemimi.moe>
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from . import ainari_request
from .access_context import AccessContext


def get_quota(context: AccessContext,
              user_id: str) -> dict:
    path = f'/v1alpha/quota/{user_id}/admin'
    return ainari_request.send_get_request(context,
                                           context.miko_address,
                                           path,
                                           "")


def list_quotas(context: AccessContext) -> dict:
    path = "/v1alpha/quota/admin"
    return ainari_request.send_get_request(context,
                                           context.miko_address,
                                           path,
                                           "")


def set_quota(context: AccessContext,
              user_id: str,
              max_instance: int,
              max_dataset: int,
              max_checkpoint: int,
              max_secret: int,
              max_taskqueue: int) -> dict:
    path = f"/v1alpha/quota/{user_id}/admin"
    json_body = {
        "max_instance": max_instance,
        "max_dataset": max_dataset,
        "max_checkpoint": max_checkpoint,
        "max_secret": max_secret,
        "max_taskqueue": max_taskqueue,
    }
    return ainari_request.send_put_request(context,
                                           context.miko_address,
                                           path,
                                           json_body)

File: src/sdk/go/ainari_sdk/quota.go
/**
 * @author      Tobias Anker <tobias.anker@kitsunemimi.moe>
 *
 * @copyright   Apache License Version 2.0
 *
 *      Copyright 2022-2026 Tobias Anker <tobias.anker@kitsunemimi.moe>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *          http://www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 */

package ainari_sdk

import (
	"fmt"
)

func GetQuota(context AccessContext, userId string) (map[string]interface{}, error) {
	path := fmt.Sprintf("v1alpha/quota/%s/admin", userId)
	vars := map[string]interface{}{}
	return SendGet(context, context.MikoAddress, path, vars)
}

func ListQuota(context AccessContext) (map[string]interface{}, error) {
	path := "v1alpha/quota/admin"
	vars := map[string]interface{}{}
	return SendGet(context, context.MikoAddress, path, vars)
}

func SetQuota(context AccessContext, userId string, maxInstance, maxDataset, maxCheckpoint, maxSecret, maxTaskqueue int) (map[string]interface{}, error) {
	path := fmt.Sprintf("v1alpha/quota/%s/admin", userId)
	jsonBody := map[string]interface{}{
		"max_instance":    maxInstance,
		"max_dataset":    maxDataset,
		"max_checkpoint": maxCheckpoint,
		"max_secret":     maxSecret,
		"max_taskqueue":  maxTaskqueue,
	}
	return SendPut(context, context.MikoAddress, path, jsonBody)
}
